rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,776 @@
+        txn_version: i64,
+        txn_timestamp: chrono::NaiveDateTime,
+        address_to_primary_name_record: &AddressToPrimaryNameRecord,
+        conn: &mut PgPoolConnection,
+    ) -> anyhow::Result<(
+        HashMap<CurrentAnsLookupPK, Self>,
+        HashMap<AnsLookupPK, AnsLookup>,
+    )> {
+        let mut new_current_ans_lookups = HashMap::new();
+        let mut new_ans_lookups = HashMap::new();
+
+        let table_item_data = write_table_item.data.as_ref().unwrap();
+        let key_type = table_item_data.key_type.as_str();
+        let key = serde_json::from_str(&table_item_data.key).unwrap_or_else(|e| {
+            panic!(
+                "version {} failed! failed to parse type {}, data {:?}. Error: {:?}",
+                txn_version, key_type, table_item_data.key, e
+            )
+        });
+        let value_type: &str = table_item_data.value_type.as_str();
+        let value = &table_item_data.value;
+        let primary_name_record: NameRecordKeyV1 =
+            serde_json::from_str(value).unwrap_or_else(|e| {
+                panic!(
+                    "version {} failed! failed to parse type {}, data {:?}. Error: {:?}",
+                    txn_version, value_type, value, e
+                )
+            });
+
+        // Case: when a primary name points to a new address, we need to set the previous primary name record's to non-primary.
+        // Lookup the previous primary name record by target address. It may not exist.
+        if let Some(mut previous_primary_record) =
+            Self::lookup_name_record_by_primary_address(address_to_primary_name_record, key, conn)
+        {
+            // Update the old primary name record
+            previous_primary_record.is_primary = false;
+            previous_primary_record.last_transaction_version = txn_version;
+            new_current_ans_lookups.insert(
+                (
+                    previous_primary_record.domain.clone(),
+                    previous_primary_record.subdomain.clone(),
+                ),
+                previous_primary_record.clone(),
+            );
+            new_ans_lookups.insert(
+                (
+                    txn_version,
+                    previous_primary_record.domain.clone(),
+                    previous_primary_record.subdomain.clone(),
+                ),
+                AnsLookup {
+                    transaction_version: txn_version,
+                    domain: previous_primary_record.domain,
+                    subdomain: previous_primary_record.subdomain,
+                    registered_address: previous_primary_record.registered_address,
+                    expiration_timestamp: previous_primary_record.expiration_timestamp,
+                    token_name: previous_primary_record.token_name,
+                    is_primary: false,
+                    is_deleted: previous_primary_record.is_deleted,
+                    transaction_timestamp: txn_timestamp,
+                },
```

Comment:
why are we using data from previous primary record? Shouldn't this be fetched from the current transaction for the new item?

Reply:
oh I see, the variable name is kinda weird. Can we you name to ans_lookups? the new_ prefix is misleading.